src/com/opc_ua/opcua_local_handler.cpp

```diff
@@ -323,15 +334,14 @@ void COPC_UA_Local_Handler::registerWithLds(const UA_String *paDiscoveryUrl) {
   }
 
   // will be freed when removed from list
-  UA_String *discoveryUrlChar = 0;
-  UA_String_copy(paDiscoveryUrl, discoveryUrlChar);
+  char *discoveryUrlChar = (char*)UA_malloc(paDiscoveryUrl->length + 1);
+  memcpy(discoveryUrlChar, paDiscoveryUrl->data, paDiscoveryUrl->length);
+  discoveryUrlChar[paDiscoveryUrl->length] = 0;
 
-  mRegisteredWithLds.pushFront(discoveryUrlChar);
+  discoveryUrlStr = UA_String_fromChars(reinterpret_cast<const char*>(discoveryUrlChar));
+  mRegisteredWithLds.pushFront(&discoveryUrlStr);
   DEVLOG_INFO("[OPC UA LOCAL]: Registering with LDS '%.*s'\n", paDiscoveryUrl->length, paDiscoveryUrl->data);
-  UA_StatusCode retVal = UA_Server_addPeriodicServerRegisterCallback(mUaServer, 0, reinterpret_cast<const char*>(discoveryUrlChar->data), 10 * 60 * 1000, 500, 0);
```

Comment:
Without a UA client, this UA_Server_addPeriodicServerRegisterCallback only make FORTE crash.